torchdynamo/variables/builder.py

```diff
@@ -226,6 +228,11 @@ def _wrap(self, value):
                 value,
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
+        elif value is List:
+            return TypingVariable(
+                value,
+                guards=make_guards(GuardBuilder.FUNCTION_MATCH),
+            )
```

Comment:
2 questions:

1) Should we follow the isintance pattern used in the other checks?
2) FUNCTION_MATCH is just:

```
if guard.is_local():
            return self.ID_MATCH(guard)
```
If you care about `guard.is_local()`, that's fine, but if not, we should just use `ID_MATCH`.

Aside: If we use `FUNCTION_MATCH` everywhere to actually mean `is_local` + `ID_MATCH`, maybe we should just rename it to `LOCAL_ID_MATCH`?

Reply:
`ID_MATCH` sounded better for this one. `typing.List` is not really a function. Thanks for pointing it out.